src-tauri/src/main.rs

```diff
@@ -7,24 +7,81 @@ mod mesh;
 mod state;
 
 use std::sync::Arc;
-use tauri::async_runtime;
+use tauri::{async_runtime, Manager};
+
+fn handle_cli_matches(
+    app: &mut tauri::App,
+    inital_autoconnect_state: &mut state::AutoConnectState,
+) -> Result<(), String> {
+    match app.get_cli_matches() {
+        Ok(matches) => {
+            let args = matches.args;
+
+            // Check if user has specified a port name to automatically connect to
+            // If so, store it for future connection attempts
+            if let Some(port_arg) = args.get("port") {
+                if port_arg.occurrences == 0 {
+                    return Ok(());
+                }
+
+                if let serde_json::Value::String(port_name) = port_arg.value.clone() {
+                    *inital_autoconnect_state = state::AutoConnectState {
+                        inner: Arc::new(async_runtime::Mutex::new(Some(port_name))),
+                    };
+                }
+            }
+
+            Ok(())
+        }
+        Err(err) => {
+            eprintln!("Error parsing CLI arguments: {}", err);
+            Err(err.to_string())
+        }
+    }
+}
 
 fn main() {
+    let initial_connection_state = state::ActiveSerialConnection {
+        inner: Arc::new(async_runtime::Mutex::new(None)),
+    };
+
+    let initial_device_state = state::ActiveMeshDevice {
+        inner: Arc::new(async_runtime::Mutex::new(None)),
+    };
+
+    let initial_graph_state = state::NetworkGraph {
+        inner: Arc::new(async_runtime::Mutex::new(None)),
+    };
+
+    let initial_analytics_state = state::AnalyticsState {
+        inner: Arc::new(async_runtime::Mutex::new(None)),
+    };
+
+    let mut inital_autoconnect_state = state::AutoConnectState {
+        inner: Arc::new(async_runtime::Mutex::new(None)),
+    };
+
     tracing_subscriber::fmt::init();
     tauri::Builder::default()
-        .manage(state::ActiveSerialConnection {
-            inner: Arc::new(async_runtime::Mutex::new(None)),
-        })
-        .manage(state::ActiveMeshDevice {
-            inner: Arc::new(async_runtime::Mutex::new(None)),
-        })
-        .manage(state::NetworkGraph {
-            inner: Arc::new(async_runtime::Mutex::new(None)),
-        })
-        .manage(state::AnalyticsState {
-            inner: Arc::new(async_runtime::Mutex::new(None)),
+        .setup(|app| {
+            match handle_cli_matches(app, &mut inital_autoconnect_state) {
+                Ok(_) => {}
+                Err(err) => panic!("Failed to parse CLI args:\n{}", err),
```

Comment:
I should have clarified in a comment, the only time this will fail is if there's a misconfigured config file. Otherwise any given argument will just return 0 occurences.